src/packages.js

```diff
@@ -451,6 +456,33 @@ export default class Packages {
       return false;
     });
   }
+  /**
+   * Preloads background files of a set of packages
+   * @param {Object[]|string[]} files Files list
+   * @param {string} type Type of files to return
+   * @return {string[]}
+   */
+  _getPreloadFileNames(files, type) {
```

Comment:
I think maybe it's a good idea to split out this method into a utility function in src/utils/packages.js instead.

Reply:
what's your suggestion for their names? 😅

Reply:
The one from `addPackages` could be named something like `constructManifest` and the `_getPreloadFileNames`  one maybe `fileNamesFromPreload` ?!

Reply:
Maybe not `constructManifest`, but `createManifestFromArray` to be a bit more explicit :nerd_face: 

Maybe you have some better suggestions ?

Reply:
It's perfect 😁

Reply:
Maybe `fileNamesFromPreload` is not precise. We want to filter `files` by their `type` and return their `filename`s. Something like `getFileNamesByType`?

Reply:
I generally only use verbs like "get" or "set" on class members. But in this case I can't really think of anything outside maybe `preloadFilenamesByType` :thinking:

Reply:
`preloadFilenamesByType` is like it's preloading something. How about `filesToFileNames`?

Reply:
Why not just embrace functions a bit more and introduce a higher-order-function for the filtering? :blush: 

```
.map(filterPreloadFilesByType('type'))
.map(preloadFilesToFilename)
```

Reply:
It will be like this:

```javascript
filenames = preloadFilesToFilename(filterPreloadFilesByType('type'))
```

Also, since it contains both "preload" and "background" types now, isn't it better to call them `files` instead of `preloadFiles`? So:

```javascript
filenames = filesToFilename(filterFilesByType('type'))
```

BTW, IMO it's better to be a single function because it's being used in 3 places.

Reply:
> Also, since it contains both "preload" and "background" types now, isn't it better to call them files instead of preloadFiles? So:

Yeah. That'll probably do fine... it's not like this conflicts with any other exported functions, and it is "namespaced" with the filename.

But if you want to be really explicit I'm open to calling it maybe `metadataFiles`.

> BTW, IMO it's better to be a single function because it's being used in 3 places.

Okay. Let's see how this looks when you change the function names :blush:

Reply:
Naming is really the hardest thing in programming😂 Is it fine?

```javascript
filenames = metadataFilesToFilenames(filterMetadataFilesByType('type'))
```

Reply:
> Naming is really the hardest thing in programming

Yeah... I've wasted so many hours (and energy) of my life by naming everything "perfectly". :notes: But in the end it doesn't even matter (insert meme here)! :notes: 

> Is it fine?

Yeah. This looks great!